R/msgmerge.R

```diff
@@ -1,16 +1,23 @@
 # split off from tools::update_pkg_po() to only run the msgmerge & checkPoFile steps
-run_msgmerge = function(po_file, pot_file) {
-  if (system(sprintf("msgmerge --update %s %s", po_file, shQuote(pot_file))) != 0L) {
+
+# https://www.gnu.org/software/gettext/manual/html_node/msgmerge-Invocation.html
+run_msgmerge <- function(po_file, pot_file, previous = FALSE, verbose = TRUE) {
+  args <- c(
+    "--update", shQuote(path.expand(po_file)),
+    if (previous) "--previous", #show previous match for fuzzy matches
+    shQuote(path.expand(pot_file))
+  )
+
+  val <- system2("msgmerge", args, stdout = TRUE, stderr = TRUE)
+  if (!identical(attr(val, "status", exact = TRUE), NULL)) {
     # nocov these warnings? i don't know how to trigger them as of this writing.
-    warningf("Running msgmerge on '%s' failed.", po_file)
+    warningf("Running msgmerge on './po/%s' failed:\n  %s", basename(po_file), paste(val, collapse = "\n"))
+  } else if (verbose) {
+    messagef("Running msgmerge on './po/%s' succeeded:\n  %s", basename(po_file), paste(val, collapse = "\n"))
   }
 
-  res <- tools::checkPoFile(po_file, strictPlural = TRUE)
-  if (nrow(res)) {
-    warningf("tools::checkPoFile() found some issues in %s", po_file)
```

Comment:
any reason to remove this warning? IINM `tools::checkPoFile()` doesn't have any incidental effect so this call will now do nothing